### PR TITLE
Extend `--artifact` API schema to allow for manifest generation

### PR DESCRIPTION
**Pull request by MarlonGamez**
Fixes: #6249, #6248 

**Description**
CC: @briandealwis @j-windsor

Users can now generate manifests using the `--artifact` (or `-a`) flag by passing some additional information in their json objects.

The schema changes are as follows:
Before: 
```json
{
  "builder": string,
  "payload": {
    "path": string,
  },
  "image": string,
},
```

After:
```json
{
  "builder": string,
  "payload": {
    "path": string,
  },
  "image": string,
  "manifest-info:": {
    "path": string,
    "port": int,
  },
}
```

**Follow-up Work**
Document this feature

### CODE REVIEW

File: pkg/skaffold/initializer/build/cli.go [hunk 1]
```diff
@@ -48,17 +52,55 @@ func (c *cliBuildInitializer) ProcessImages(images []string) error {
...
 func (c *cliBuildInitializer) BuildConfig() (latestV1.BuildConfig, []*latestV1.PortForwardResource) {
+	pf := []*latestV1.PortForwardResource{}
+
+	for _, manifestInfo := range c.manifests {
+		// Port value is set to 0 if user decides to not port forward service
+		if manifestInfo.Port != 0 {
+			pf = append(pf, &latestV1.PortForwardResource{
+				Type: "service",
+				Name: manifestInfo.Name,
+				Port: util.FromInt(manifestInfo.Port),
```
Comment (nkubala) [inline pkg/skaffold/initializer/build/cli.go:63]: what happens if two manifests specify the same local port? should we warn the user in that case?

**Review by nkubala — COMMENTED** [review]
code generally looks good, one question about the ports

File: pkg/skaffold/initializer/build/cli.go (hunk 1 above)
Comment (MarlonGamez) [reply pkg/skaffold/initializer/build/cli.go:63]: That's a good point, I think that we should probably do this in our interactive version, but with the `--artifact` flag, it seems that this should be handled on the end of whoever is utilizing the flag.

I can open an issue to modify the interactive version

**Review by nkubala — APPROVED** [review]

File: pkg/skaffold/initializer/build/cli.go (hunk 1 above)
Comment (briandealwis) [reply pkg/skaffold/initializer/build/cli.go:63]: Only containers *within* a pod need to have distinct ports — pods have distinct port spaces.

**Pull request merged**